### ISSUE TITLE
Only seeing 1 worker in the dashboard - is that expected?

### ISSUE THREAD

**Issue by bersena911** [mludvig/aws-ethereum-miner#47]
First of all Thanks for everything you did <3

now my question: Etherminer only shows 1 worker in dashboard when actually 2 EC2 instances are running. Is it okay?

**Comment by mludvig** [comment]
Hi, yes it’s expected. The AWS workers don’t have a unique name and are all reported together under “default”. 

The hashrate should be a sum of all your workers’ hashrates (give or take a few % due to how ethermine.org calculates it).

**Comment by bersena911** [comment]
@mludvig is there any way to check how good are instances doing? currently I get 500 MH/s Hashrate. I am running

1 - g4ad.8xlarge
4 - g5.xlarge
2 - p3.2xlarge

I wonder how did it end up 500 MH/s which instances are doing well.

**Comment by bersena911** [comment]
@mludvig does it automatically launch new spot instances when it becomes available? or we have to redeploy template.

**Comment by mludvig** [comment]
Hi,

1. The performance: g4ad.8xlarge is 96 MH/s, g5.xlarge is 56 MH/s, p3.2xlarge is 93 MH/s - all up with your instances it's around 500 MH/s, so yeah, it checks.
2. Yes it will always try to reach the hashrate that you specified in the CFN Parameters. If you request for example Hashrate 1000 MH/s and it only spins up 500 MH/s at first it will keep trying to launch more over time to fulfil the 1 GH/s. You can check the ASG Activity tab to see what it's doing.
3. The ASG manages the spot fleet and creates one-off spot requests. However it keeps trying and creating more so don't worry about that, it's just a technical implementation. It acts the same as having persistent spot requests, just done in a different way that supports more options.

The bottom line is that yes it will keep replacing failed instances and keep creating new ones to fulfil the requested capacity. You don't have to redeploy.

**Comment by bersena911** [comment]
@mludvig This is Fantastic!!! Thanks.

**Comment by bornnel** [comment]
thanks for this information on this can't wait to start..
i request for this instant limit (EC2 Instances / Instance Limit (All G and VT instances), New Limit = 8)

and still, I can't create instance or start the mining software with my current limit. Please help

**Comment by mludvig** [comment]
Have they approved the limit already? If not you’ll have to wait until they do.

**Comment by bornnel** [comment]
thanks for your reply. No, they have not. they said 48 hours

**Comment by bornnel** [comment]
> Have they approved the limit already? If not you’ll have to wait until they do.

It's has been approved>
Why use metamask? please and how do I config it. "Best to use [MetaMask](https://metamask.io/)"
Its is working now. I will share some eth with you. thanks for your post.

If I wanted to make all instances just g4ad.xlarge please what would I do?

**Comment by bornnel** [comment]
Hi please, I am trying to use an auto scale to stop instances that are not performing at top hast rate. 
Please can you help?

This link did a good job: https://betterprogramming.pub/how-i-made-a-profit-mining-ethereum-on-aws-2ffe3aab579
But I cant add a parameter it's giving me an error -
------

ave the same issues

KeyPairName:
    Description: Name of EC2 key pair assigned to instances.
    Type: String
    
    
LaunchTemplate:
  Type: AWS::EC2::LaunchTemplate
  Properties:
    ...
    KeyName: !Ref KeyPairName 
01/06/2022, 12:36:55 - Template contains errors.: Template format error: YAML not well-formed. (line 39, column 3)

Please kindly help

**Comment by mludvig** [comment]
My templates don’t use KeyPairName - you must be using some modified / broken template. Where did you get it from?

**Comment by bornnel** [comment]
I got it from James Robinson post. Tried adding his parameters but it's keep breaking. 


Sent from Yahoo Mail on Android 
 
  On Wed, 1 Jun 2022 at 21:08, Michael ***@***.***> wrote:   


My templates don’t use KeyPairName - you must be using some modified / broken template. Where did you get it from?

**Comment by mludvig** [comment]
Why don't you use my template instead? It's evolved a lot since JR's post. Simply click the quick-launch link from the table here: https://github.com/mludvig/aws-ethereum-miner#quick-start (use the Default VPC links).

If you want to run it in all your regions try this: https://gist.github.com/mludvig/d21b03baf53a5499626f1a786af7b6f7

**Comment by bornnel** [comment]
Ok. Thanks, which do you recommend? spot or on-demand? I have started again?

**Comment by bornnel** [comment]
Soi tried a few days 4 days and look at my billing rate.. crazy
![Screenshot (37)](https://user-images.githubusercontent.com/1068222/171966769-754a7e88-e630-40a1-b0e0-6f49093c8e91.png)

I don't know what to do next...

**Comment by mludvig** [comment]
@bornnel You're running a bunch of EC2 instances and are being charged for it. What exactly is the problem you're seeing?

**Comment by bornnel** [comment]
> @bornnel You're running a bunch of EC2 instances and are being charged for it. What exactly is the problem you're seeing?
I am running the VPC stack on two different regions. and for less than 10 hours I was saying 48 hours.  called the support they said its charges instances for 1hours even when you run it for 2mins.
